vector/src/main/java/im/vector/app/VectorApplication.kt

```diff
@@ -220,16 +218,9 @@ class VectorApplication :
         }
     }
 
-    override fun providesMatrixConfiguration(): MatrixConfiguration {
```

Comment:
the main downside to using `MatrixConfiguration.Provider` is that it occurs during `super.onCreate` which means the dependencies declared in the application haven't been injected yet, making the object creation lifecycle a bit misleading